### PR TITLE
[EVNT-261] - Restrict Integrations to accept only https

### PR DESCRIPTION
**Pull request by jpramos123**
[EVNT-261](https://issues.redhat.com/browse/EVNT-261)

Remove HTTP handling. Only accepts HTTPS

### CODE REVIEW

File: splunk-quarkus/src/main/java/com/redhat/console/notifications/splunkintegration/SplunkIntegration.java [hunk 1]
```diff
@@ -208,19 +208,10 @@ private void configureHandler() throws Exception {
...
                 .setHeader(Exchange.HTTP_PATH, constant("/services/collector/event"))
-                .choice()
-                .when(simple("${headers.metadata[url]} startsWith 'http://'"))
-                .to(http("dynamic")
-                        .httpMethod("POST")
-                        .advanced()
-                        .httpClientConfigurer(getClientConfigurer()))
-                .endChoice()
-                .otherwise()
                 .to(https("dynamic")
```
[the remark this answers is not in the document]
Comment (jpramos123) [reply splunk-quarkus/src/main/java/com/redhat/console/notifications/splunkintegration/SplunkIntegration.java:211]: we get:

```
Failed cloud event, id 9dc9a4b1-8868-4afc-a69d-e8723b20452c, with HTTP exception : HTTP operation failed invoking http://2391-187-3-29-242.ngrok.io/services/collector/event with statusCode: 503
```

[the remark this answers is not in the document]
Comment (beav) [reply splunk-quarkus/src/main/java/com/redhat/console/notifications/splunkintegration/SplunkIntegration.java:211]: Is the 503 coming from the ngrok service? if so, it means it tried to connect w/ http I think

[the remark this answers is not in the document]
Comment (vkrizan) [reply splunk-quarkus/src/main/java/com/redhat/console/notifications/splunkintegration/SplunkIntegration.java:211]: Yeah, that's what it looks like. 

I wonder if we can fail early if it is not `https://`. @jpramos123 any suggestions?

[the remark this answers is not in the document]
Comment (vkrizan) [reply splunk-quarkus/src/main/java/com/redhat/console/notifications/splunkintegration/SplunkIntegration.java:211]: There is `throwException()` that can be used. https://camel.apache.org/manual/route-configuration.html#_assigning_route_configurations_to_routes

[the remark this answers is not in the document]
Comment (jpramos123) [reply splunk-quarkus/src/main/java/com/redhat/console/notifications/splunkintegration/SplunkIntegration.java:238]: @vkrizan @beav Added a new commit with the error handling and early validation of `http://`.

Let me know if this works.

Next is to validate what @beav suggested

Thanks

File: splunk-quarkus/src/main/java/com/redhat/console/notifications/splunkintegration/SplunkIntegration.java [hunk 2]
```diff
@@ -131,6 +136,20 @@ private void configureHttpFailed() throws Exception {
                 .to(direct("return"));
     }
 
+    private void configureSecureConnectionFailed() throws Exception {
+        Processor ceEncoder = new CloudEventEncoder(COMPONENT_NAME, RETURN_TYPE);
+        Processor resultTransformer = new ResultTransformer();
+        // The error handler when we receive an HTTP (unsecure) connection instead of HTTPS
+        from(direct("secureConnectionFailed"))
+                .setBody(simple("Unsecure connection detected"))
```
Comment (vkrizan) [inline splunk-quarkus/src/main/java/com/redhat/console/notifications/splunkintegration/SplunkIntegration.java:144]: `s/Unsecure/Insecure/`

File: splunk-quarkus/src/main/java/com/redhat/console/notifications/splunkintegration/SplunkIntegration.java [hunk 3]
```diff
@@ -131,6 +136,20 @@ private void configureHttpFailed() throws Exception {
...
+                .setHeader("outcome-fail", simple("true"))
+                .process(resultTransformer)
+                .marshal().json()
+            .log(LoggingLevel.ERROR, "Unsecure connection detected, id ${header.ce-id}")
+            .process(ceEncoder)
```
Comment (vkrizan) [inline splunk-quarkus/src/main/java/com/redhat/console/notifications/splunkintegration/SplunkIntegration.java:149]: I've noticed that this encoder process can be moved completely to "return" route.

File: splunk-quarkus/src/main/java/com/redhat/console/notifications/splunkintegration/SplunkIntegration.java [hunk 4]
```diff
@@ -208,20 +235,10 @@ private void configureHandler() throws Exception {
...
-                        .advanced()
-                        .httpClientConfigurer(getClientConfigurer()))
-                .endChoice()
-                .otherwise()
                 .to(https("dynamic")
                         .httpMethod("POST")
                         .advanced()
                         .httpClientConfigurer(getClientConfigurer()))
-                .endChoice()
-                .end()
```
Comment (vkrizan) [inline splunk-quarkus/src/main/java/com/redhat/console/notifications/splunkintegration/SplunkIntegration.java:224]: Can these two be kept and moved to bottom? It should avoid future bugs.

**Review by vkrizan — COMMENTED** [review]
Looks better.

The increasing number of error handler routes raises a question of refactoring them to a single one with parameters.

@jpramos123 please also run code formatter with `mvnw formatter:format`.

File: splunk-quarkus/src/main/java/com/redhat/console/notifications/splunkintegration/SplunkIntegration.java (hunk 2 above)
Comment (jpramos123) [reply splunk-quarkus/src/main/java/com/redhat/console/notifications/splunkintegration/SplunkIntegration.java:144]: Thanks!

File: splunk-quarkus/src/main/java/com/redhat/console/notifications/splunkintegration/SplunkIntegration.java (hunk 3 above)
Comment (jpramos123) [reply splunk-quarkus/src/main/java/com/redhat/console/notifications/splunkintegration/SplunkIntegration.java:149]: Moved it to `return` route.

Thanks

File: splunk-quarkus/src/main/java/com/redhat/console/notifications/splunkintegration/SplunkIntegration.java (hunk 4 above)
Comment (jpramos123) [reply splunk-quarkus/src/main/java/com/redhat/console/notifications/splunkintegration/SplunkIntegration.java:224]: Keeping `.end()`. I can't keep `.endChoice` as `.choice` was removed in line 211.

**Comment by petrblaho** [comment]
Created different implementation https://github.com/RedHatInsights/eventing-integrations/pull/67 using some of the code from this PR

**Pull request closed without merging**